src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -51,34 +67,87 @@ public void initializeConverter() {
 
         clusterColorControl = (ZclColorControlCluster) endpoint.getInputCluster(ZclColorControlCluster.CLUSTER_ID);
         if (clusterColorControl == null) {
-            logger.error("Error opening device control controls {}", endpoint.getIeeeAddress());
+            logger.error("Error opening device color controls {}", endpoint.getIeeeAddress());
             return;
         }
 
         clusterLevelControl = (ZclLevelControlCluster) endpoint.getInputCluster(ZclLevelControlCluster.CLUSTER_ID);
         if (clusterLevelControl == null) {
-            logger.error("Error opening device level controls {}", endpoint.getIeeeAddress());
-            return;
+            logger.warn("Device does not support level control {}", endpoint.getIeeeAddress());
         }
 
-        clusterColorControl.bind();
-        clusterLevelControl.bind();
+        clusterOnOff = (ZclOnOffCluster) endpoint.getInputCluster(ZclOnOffCluster.CLUSTER_ID);
+        if (clusterOnOff == null) {
+            logger.warn("Device does not support on/off control {}", endpoint.getIeeeAddress());
+        }
 
-        // Add a listener, then request the status
+        // Discover whether the device supports HUE/SAT or XY color set of commands
+        try {
+            if (!clusterColorControl.discoverAttributes(false).get()) {
+                logger.warn("{}: Cannot determine whether device supports RGB color. Assuming it supports HUE/SAT",
+                        endpoint.getIeeeAddress());
+                supportsHue = true;
+            } else if (clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTHUE)) {
+                logger.debug("Device supports Hue/Saturation color set of commands {}", endpoint.getIeeeAddress());
+                supportsHue = true;
+            } else if (clusterColorControl.getSupportedAttributes().contains(ZclColorControlCluster.ATTR_CURRENTX)) {
+                logger.debug("Device supports XY color set of commands {}", endpoint.getIeeeAddress());
+                supportsXY = true;
+            } else {
+                logger.warn("Device does not support RGB color {}", endpoint.getIeeeAddress());
```

Comment:
I am warning here because we arrived at this point of code assuming the device supports color, and it doesn't. So we created the channel and it will not work. Should'nt we warn instead of debug here ?

Reply:
Ok.